apps/daemon/pkg/toolbox/process/pty/controller.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2025 Daytona Platforms Inc.
+// SPDX-License-Identifier: AGPL-3.0
+
+package pty
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/gorilla/websocket"
+	log "github.com/sirupsen/logrus"
+)
+
+// NewPTYController creates a new PTY controller
+func NewPTYController(workDir string) *PTYController {
+	return &PTYController{workDir: workDir}
+}
+
+// CreatePTYSession creates a new PTY session
+func (p *PTYController) CreatePTYSession(c *gin.Context) {
+	var req PTYCreateRequest
+	if err := c.ShouldBindJSON(&req); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	// Validate session ID
+	if req.ID == "" {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "session ID is required"})
+		return
+	}
+
+	// Check if session with this ID already exists
+	if _, exists := ptyManager.Get(req.ID); exists {
+		c.JSON(http.StatusConflict, gin.H{"error": fmt.Sprintf("PTY session with ID '%s' already exists", req.ID)})
+		return
+	}
+
+	// Defaults
+	if req.Cwd == "" {
+		req.Cwd = p.workDir
+	}
+	if req.Envs == nil {
+		req.Envs = make(map[string]string, 1)
+	}
+	if req.Envs["TERM"] == "" {
+		req.Envs["TERM"] = "xterm-256color"
+	}
+	if req.Cols <= 0 {
```

Comment:
I suggest we throw `BadRequest` errors instead of overriding what the user sends.

The user sends their expected params and we should throw if we can't respect it.